src/colorfulness.jl

```diff
@@ -0,0 +1,57 @@
+using ImageCore: GenericGrayImage
+using Statistics: mean, std
+
+"""
+
+    HASLER_AND_SUSSTRUNK_M3 <: NoReferenceIQI
+    M =  hasler_and_susstrunk_m3(img)
+    M =  colorfulness(img)
+
+Calculates the colorfulness of an RGB image according to the metric,
+M3 from [1]. As a guide to interpretation of results, the authors
+suggest:
+
+|Not colorful        |  0|
+|slightly colorful   | 15|
+|moderately colorful | 33|
+|averagely colorful  | 45|
+|quite colorful      | 59|
+|highly colorful     | 82|
+|extremely colorful  |109|
+
+[1] Hasler, D. and Süsstrunk, S.E., 2003, June. Measuring colorfulness
+in natural images. In Human vision and electronic imaging VIII
+(Vol. 5007, pp. 87-96). International Society for Optics and
+Photonics.
+
+"""
+struct HASLER_AND_SUSSTRUNK_M3 <: NoReferenceIQI end
+
+# api
+(iqi::HASLER_AND_SUSSTRUNK_M3)(img) = hasler_and_susstrunk_m3(img)
+
+function hasler_and_susstrunk_m3(img::AbstractArray{<:AbstractRGB})
+
+    R = 255 .* float(red.(img))
```

Comment:
I believe these `255`s are deliberately used, right?

Reply:
Yes - unfortunately the metric uses [0,255] values for RGB not [0,1] as per Julia Images. I think we have to be faithful to the paper so that the interpretation of the metric is consistent.